ios/RCTWebRTC/WebRTCModule+RTCPeerConnection.m

```diff
@@ -327,34 +331,40 @@ - (void)peerConnection:(RTCPeerConnection *)peerConnection didChangeSignalingSta
 - (void)peerConnection:(RTCPeerConnection *)peerConnection didAddStream:(RTCMediaStream *)stream {
   NSMutableArray *tracks = [NSMutableArray array];
   for (RTCVideoTrack *track in stream.videoTracks) {
-    NSString *tag = [self reactTagForID:track.trackId RTCPC:peerConnection];
-    self.remoteTracks[tag] = track;
+    peerConnection.remoteTracks[track.trackId] = track;
     [tracks addObject:@{@"id": track.trackId, @"kind": track.kind, @"label": track.trackId, @"enabled": @(track.isEnabled), @"remote": @(YES), @"readyState": @"live"}];
   }
   for (RTCAudioTrack *track in stream.audioTracks) {
-    NSString *tag = [self reactTagForID:track.trackId RTCPC:peerConnection];
-    self.remoteTracks[tag] = track;
+    peerConnection.remoteTracks[track.trackId] = track;
     [tracks addObject:@{@"id": track.trackId, @"kind": track.kind, @"label": track.trackId, @"enabled": @(track.isEnabled), @"remote": @(YES), @"readyState": @"live"}];
   }
 
-  NSString *streamReactTag =  [self reactTagForID:stream.streamId RTCPC:peerConnection];
-  self.remoteStreams[streamReactTag] = stream;
+  NSString *streamReactTag;
+  // Make sure ID does not exist across local and remote streams (for any peerConnection)
```

Comment:
It won't because this access is necessary anyway for the view manager which needs to obtain stream for react tag. It needs to search all peerconnections an local streams.